source/Pacioli/Pacioli.Lib/Models/JournalEntry.cs

```diff
@@ -13,15 +13,29 @@ public JournalEntry()
             Credits = new List<JournalEntryItem>();
         }
 
-        public JournalEntry([NotNull] List<JournalEntryItem> debits, [NotNull] List<JournalEntryItem> credits)
+        public JournalEntry([NotNull]Account account, [NotNull] DateTime date, [NotNull] List<JournalEntryItem> debits, [NotNull] List<JournalEntryItem> credits)
         {
-            if (debits is null || credits is null)
+            if (account is null)
+                throw new ArgumentNullException("Account should not be null");
+
+            if (date == DateTime.MinValue)
+                throw new ArgumentException("Please specify a valid date");
+
+            if (debits is null || credits is null || account is null || date == DateTime.MinValue)
```

Comment:
Value types can't be null, you're already checking `date == DateTime.MinValue` in the above statement.

Reply:
You can make an array of referenced types and check them for null with Linq
```CSharp
var referenceParams = new object[] {debits, credits, account};
            if (referenceParams.Any(param => param is null))
                throw new ArgumentNullException("Debits, credits and account should not be null");
```